### PR TITLE
Fix override keyword being print to the left side

### PR DESCRIPTION
**Pull request by giulianobelinassi**
Previously, the `override` keyword in C++ was being print in the left side of a method decl, which is unsupported by C++ standard. This commit fixes that by setting the `CanPrintOnLeft` field to 0, forcing it to be print on the right side of the decl.

This comes from the fact that #87281 won't be backported into clang-18. However since classes with `override` keyword is broken in LLVM-18, I think this is still worth fixing.

@vgvassilev Please check if it works in your test cases.

Notice that this PR targets release/18.x, and not the main branch.

### CODE REVIEW

**Review by AaronBallman — CHANGES_REQUESTED** [review]
I don't think we should add this to the 18.x release -- we're already getting requests from users to not do that: https://lists.llvm.org/pipermail/cfe-commits/Week-of-Mon-20240408/566402.html

**Comment by giulianobelinassi** [comment]
> I don't think we should add this to the 18.x release -- we're already getting requests from users to not do that: https://lists.llvm.org/pipermail/cfe-commits/Week-of-Mon-20240408/566402.html

This doesn't affect the number of spaces and it is not based on vgvassilev's patch.

The test case in question:
```
~/projects/llvm-project/llvm/build/bin/clang++ a.cpp -Xclang -ast-print -S -o -
template <typename T> class FinalTemplate;
```

This would only change the position of the `override` and `final` keywords.

**Comment by AaronBallman** [comment]
I think the 18.1 behavior is already correct?

https://godbolt.org/z/v3G7h44E1

**Comment by giulianobelinassi** [comment]
> I think the 18.1 behavior is already correct?
> 
> https://godbolt.org/z/v3G7h44E1

It is not:
https://godbolt.org/z/96rWese3P

**Comment by AaronBallman** [comment]
Ah that's right, it was with definitions that we ran into the issue! Sorry for being slow today. Okay, I retract my concerns, this is a safe and targeted fix.

**Review by AaronBallman — COMMENTED** [review]
I think this is reasonable, but leaving it to @erichkeane to make the final call.

**Review by erichkeane — APPROVED** [review]

**Comment by giulianobelinassi** [comment]
The branch is marked as ` Changes requested ` but no change was pointed in the code. Is everything correct?

**Comment by Sirraide** [comment]
> The branch is marked as `Changes requested` but no change was pointed in the code. Is everything correct?

That’s just because Aaron didn’t submit another review, so github still marks it as ‘changes requested’; seeing as it has been approved, the rest is now up to the release manager.

**Review by AaronBallman — APPROVED** [review]
Sorry about leaving it in a "requests changes" state, LGTM!

**Comment by tstellar** [comment]
Hi @giulianobelinassi (or anyone else). If you would like to add a note about this fix in the release notes (completely optional). Please reply to this comment with a one or two sentence description of the fix.

**Pull request merged**
